client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -160,31 +163,17 @@ class CalypsoifyIframe extends Component<
 	mediaCancelPort: MessagePort | null = null;
 	revisionsPort: MessagePort | null = null;
 	checkoutPort: MessagePort | null = null;
-	successfulIframeLoad = false;
-	waitForIframeToInit: ReturnType< typeof setInterval > | undefined = undefined;
-	waitForIframeToLoad: ReturnType< typeof setTimeout > | undefined = undefined;
 
 	componentDidMount() {
 		window.addEventListener( 'message', this.onMessage, false );
 
-		const isDesktop = config.isEnabled( 'desktop' );
-		// If the iframe fails to load for some reson, eg. an unexpected auth loop, this timeout
-		// provides a redirect to wpadmin for web users - this should now be a rare occurance with
-		// a 3rd party cookie auth issue fix in place https://github.com/Automattic/jetpack/pull/16167
-		this.waitForIframeToInit = setInterval( () => {
-			if ( this.props.shouldLoadIframe ) {
-				clearInterval( ( this.waitForIframeToInit as unknown ) as number );
-				this.waitForIframeToLoad = setTimeout( () => {
-					isDesktop
-						? this.props.notifyDesktopCannotOpenEditor(
-								this.props.site,
-								REASON_BLOCK_EDITOR_UNKOWN_IFRAME_LOAD_FAILURE,
-								this.props.iframeUrl
-						  )
-						: window.location.replace( this.props.iframeUrl );
-				}, 25000 );
-			}
-		}, 1000 );
+		// If the iframe itself fails to load for some reason, eg. an unexpected
+		// auth loop, we need to exit to wp-admin without the iFrame. Therefore,
+		// once we start loading the iFrame, we set the editor redirect timer to
+		// 25s. This timer is cleared when iFrame.onLoad is called.
+		if ( this.props.shouldLoadIframe && ! this.editorRedirectTimer ) {
+			this.setEditorRedirectTimer( 25000 );
```

Comment:
@glendaviesnz @scruffian @aaronrobertshaw Could you perhaps comment on these changes? Do we still need to use this lengthy redirect approach or have the underlying auth issues been solved?

The main change is instead of having multiple timers, I've implemented one timer:
1. up to 25 seconds until iframe.onload is called
2. a further 1 second until the iframe bridge server sends us a response

What do you think?

Reply:
Certainly wouldn't hurt to simplify this code if possible. From memory some of these checks were added to handle the loading of Atomic sites with custom domains, and [jetpack sites with iframe headers set](https://github.com/Automattic/wp-calypso/pull/36977), so any changes would need testing against these contexts rather than just testing that simple sites load the iframed editor.

Reply:
Thanks for the context! I just tested with a JN site, and it looks like the X-Frame redirect would happen after `iframe.onload` is called. Additionally, the auth redirect happens at a higher layer of the stack before trying to load the iframe component. (Ran into that when authenticating and setting up wpcom SSO)

<img width="1668" alt="Screen Shot 2021-04-13 at 2 17 41 PM" src="https://user-images.githubusercontent.com/6265975/114622620-100e4c80-9c63-11eb-9365-52b0d9350824.png">

I think this makes sense to me because `iframe.onload` is called when the frame's sources have finished loading. So if the browser shows an error page like above, the frame still considers it as loading a page. Which is kind of ridiculous, but that seems to be pretty consistent behavior, and matches articles I'm reading about cross-origin iFrames.

I'll do some testing with custom Atomic domains too, but if this holds up, we can probably remove the 25 second redirect timer and just utilize `iframe.onload`

Reply:
Another benefit of that is `iframe.onload` fires only when all scripts and content have been downloaded by the frame, so we wouldn't need an extra long timeout to handle network connections

Reply:
Though, looking at #43248, it might be important to keep for the auth redirect problem 🤔 

Though if that problem is solved, we won't need it either

Reply:
Yep, I'll just keep the long redirect around to handle cases like that. I'll try to be clear in the comments that that's a more rare condition